crates/ra_ide/src/hover.rs

```diff
@@ -18,14 +18,64 @@ use ra_syntax::{
 };
 
 use crate::{
-    display::{macro_label, rust_code_markup, rust_code_markup_with_doc, ShortLabel},
-    FilePosition, RangeInfo,
+    display::{macro_label, rust_code_markup, rust_code_markup_with_doc, ShortLabel, ToNav},
+    runnables::runnable,
+    FileId, FilePosition, NavigationTarget, RangeInfo, Runnable,
 };
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct HoverConfig {
+    pub run: bool,
+    pub debug: bool,
+    pub goto_type_def: bool,
+    pub implementations: bool,
+}
+
+impl Default for HoverConfig {
+    fn default() -> Self {
+        // A client should explicitly specify if it supports hover actions!
```

Comment:
I think in `ra_ide` we generally enable everything by deafault, and than `rust-analyzer` explicitelly disables stuff it doesn't like.

Reply:
ok. Besides, a I've added a hoverActions client capability, so this initial defaults outdated.